File: programmers/lev2/1845.py
def solution(nums):
    answer = 0
    num_dict=dict()
    sel_num=len(nums)//2

    for i in nums:
        num_dict[i]=num_dict.get(i,0)+1
    
    if sel_num>=len(num_dict.keys()):
        answer=len(num_dict.keys())
    else:
        answer=sel_num
    return answer

nums=[3,1,2,3]
solution(nums)

File: java/src/programmers/year2023/lv1/_1845.java
package programmers.year2023.lv1;

import java.util.HashMap;

public class _1845 {
    public static void main(String[] args) {
        int[] nums = {3, 1, 2, 3};

        System.out.println(solution(nums));

    }

    public static int solution(int[] nums) {
        int answer = 0;

        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int now = nums[i];
            if (!map.containsKey(now)) {
                map.put(now, 1);
            }
        }

        answer = map.keySet().size() <= nums.length / 2 ? map.keySet().size() : nums.length / 2;

        return answer;
    }
}
